src/OrchardCore/OrchardCore.DisplayManagement/ShapeFactoryExtensions.cs

```diff
@@ -0,0 +1,195 @@
+    {
+        if (parameters == null || parameters == Arguments.Empty)
+        {
+            return factory.CreateAsync(shapeType);
+        }
+
+        return factory.CreateAsync(shapeType, _newShape, null, createdContext =>
+        {
+            var shape = (Shape)createdContext.Shape;
+
+            // If only one non-Type, use it as the source object to copy
+
+            var initializer = parameters.Positional.SingleOrDefault();
+
+            if (initializer != null)
+            {
+                // Use the Arguments class to optimize reflection code
+                var arguments = Arguments.From(initializer);
+
+                foreach (var prop in arguments.Named)
+                {
+                    shape.Properties[prop.Key] = prop.Value;
+                }
+            }
+            else
+            {
+                foreach (var kv in parameters.Named)
+                {
+                    shape.Properties[kv.Key] = kv.Value;
+                }
+            }
+        });
+    }
+
+    /// <summary>
+    /// Creates a dynamic proxy instance for the type and initializes it.
+    /// </summary>
+    /// <typeparam name="TModel">The type to instantiate.</typeparam>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="initializeAsync">The initialization method.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, Func<TModel, ValueTask> initializeAsync)
+        where TModel : class
+        => factory.CreateAsync(typeof(TModel).Name, initializeAsync);
+
+    /// <summary>
+    /// Creates a dynamic proxy instance for the type and initializes it.
+    /// </summary>
+    /// <typeparam name="TModel">The type to instantiate.</typeparam>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="shapeType">The shape type to create.</param>
+    /// <param name="initializeAsync">The initialization method.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, string shapeType, Func<TModel, ValueTask> initializeAsync)
+        where TModel : class
+    {
+        return factory.CreateAsync(shapeType, () => ShapeFactory(initializeAsync));
+
+        static ValueTask<IShape> ShapeFactory(Func<TModel, ValueTask> init)
+        {
+            var shape = CreateStronglyTypedShape(typeof(TModel));
+
+            if (init != null)
+            {
+                var task = init((TModel)shape);
+
+                if (!task.IsCompletedSuccessfully)
+                {
+                    return Awaited(task, shape);
+                }
+            }
+
+            return ValueTask.FromResult(shape);
+        }
+
+        static async ValueTask<IShape> Awaited(ValueTask task, IShape shape)
+        {
+            await task;
+            return shape;
```

Comment:
```suggestion

            return shape;
```

Reply:
Please second time accept the code suggestions if it's fine rather than resolve them and add them yourself